src/java/org/apache/cassandra/index/sai/disk/ByteSliceReader.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk;
+
+import org.apache.lucene.store.DataInput;
+import org.apache.lucene.util.ByteBlockPool;
+
+/* IndexInput that knows how to read the byte slices written
+ * by Posting and PostingVector.  We read the bytes in
+ * each slice until we hit the end of that slice at which
+ * point we read the forwarding address of the next slice
+ * and then jump to it.*/
```

Comment:
Nit: This could be class JavaDoc. If it uses `{@link}` annotations it will be easier to refactor and detect when it's outdated. For example, the first mentioned class, `IndexInput`, seems to actually mean `DataInput`.

Reply:
Ah, this comes from Lucene. I think it would be useful to mark the classes that are forked from Lucene, maybe in the class JavaDoc or with a custom annotation.

Reply:
@adelapena I have added a javadoc comment above the class that notes the copy of the lucene class and the reason for the copy but I don't know if we need to add any copyright notice or not to indicate that it is lucene code. Do you know if there is a precedence for this in the codebase anywhere?

Reply:
I'm not sure, but 'd say we don't need any additional copyright notice since both Cassandra and Lucene belong to the ASF. That javadoc comment about the copy of the class and the reason for the copy should be enough, although I don't see that comment on this class.